gui/src/2D/controls/toggleButton.ts

```diff
@@ -0,0 +1,308 @@
+     */
+    public pointerDownActiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer up animation when the toggle is active.
+     */
+    public pointerUpActiveAnimation: () => void;
+
+    /**
+     * Function called to generate a pointer enter animation when the toggle is inactive.
+     */
+    public pointerEnterInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer out animation when the toggle is inactive.
+     */
+    public pointerOutInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer down animation when the toggle is inactive.
+     */
+    public pointerDownInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer up animation when the toggle is inactive.
+     */
+    public pointerUpInactiveAnimation: () => void;
+
+    /** Observable raised when isActive is changed */
+    public onIsActiveChangedObservable = new Observable<boolean>();
+
+    /**
+     * Gets or sets a boolean indicating that the toggle will let internal controls handle picking instead of doing it directly using its bounding info
+     */
+    public delegatePickingToChildren = false;
+
+    private _image: Nullable<Image>;
+    /**
+     * Returns the image part of the button (if any)
+     */
+    public get image(): Nullable<Image> {
+        return this._image;
+    }
+
+    private _textBlock: Nullable<TextBlock>;
+    /**
+     * Returns the image part of the button (if any)
+     */
+    public get textBlock(): Nullable<TextBlock> {
+        return this._textBlock;
+    }
+
+    private _group: string;
+    /** Gets or sets group name this toggle belongs to */
+    public get group(): string {
+        return this._group;
+    }
+    public set group(value: string) {
+        if (this._group === value) {
+            return;
+        }
+
+        this._group = value;
+    }
+
+    private _isActive = false;
+    /** Gets or sets a boolean indicating if the toogle button is active or not */
+    public get isActive(): boolean {
+        return this._isActive;
+    }
+    public set isActive(value: boolean) {
+        // Function modeled after radioButton.ts
+        if (this._isActive === value) {
+            return;
+        }
+
+        this._isActive = value;
+
+        // Update the visual state based on the new value
+        if (this._isActive) {
+            this.toActiveAnimation();
+        } else {
+            this.toInactiveAnimation();
+        }
+
+        this._markAsDirty();
+
+        this.onIsActiveChangedObservable.notifyObservers(value);
+
+        if (this._isActive && this._host) {
+            // Update all controls from same group
+            this._host.executeOnAllControls((control) => {
+                if (control == this) {
+                    return;
+                }
+
+                if ((<any>control).group === undefined) {
```

Comment:
Maybe you can get use `_getTypeName() === "ToggleButton"` instead of testing for the `group` property (in case one day other control(s) define a `group` property)?

Reply:
I did a rewrite of this function utilizing `_getTypeName()` and tried to simplify the logic. Hopefully the comments make sense, but if not let me know and I can give you pseudocode to review.